backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -152,16 +156,225 @@ const testGetUsername = () => {
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the update data have extra properties', async () => {
+			const data = { firstName: 'newName', extra: 'extraProp' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the oldPassword is not correct', async () => {
+			const data = { oldPassword: 'invalid', newPassword: 'newPassword123.' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.incorrectPassword.status);
+			expect(res.body.code).toEqual(templates.incorrectPassword.code);
+		});
+
+		test('should fail if the update data have oldPassword but not newPassword', async () => {
+			const data = { oldPassword: testUser.password };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the update data have newPassword but not oldPassword', async () => {
+			const data = { newPassword: 'newPassword123.' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the update data have weak newPassword', async () => {
+			const data = { oldPassword: testUser.password, newPassword: 'abc' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the update data have a newPassword which is the same as the old one', async () => {
+			const data = { oldPassword: testUser.password, newPassword: testUser.password };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should update the profile if the user is logged in', async () => {
+			const data = { firstName: 'newName' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			await testSession.put('/v5/user/').send(data).expect(200);
```

Comment:
Can we do a get here to check that the name really has been updated?

Reply:
should I do that for the other tests where I update the email as well ?

Reply:
yea any PUT/POST if we can do a GET to verify that'd be good